### ISSUE TITLE
Data source

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/llm_scratch#5]
We are using the `wmt/wmt14`(https://huggingface.co/datasets/wmt/wmt14) dataset where the number of samples in the `de-en` and the `fr-en` subsets matches with the numbers reported in the paper.

## What's bugging data wise?

The attention is all you need paper used the `newstest13` development set to report all their evaluation results. Upon digging a bit I notice that they are using `newstest14` dataset as the test set and both `["newsdev2014", "newstest2013"]` as the development  set (https://huggingface.co/datasets/wmt/wmt14/blob/55df9961a21e6d05f0b4e2de059c275ee774e869/wmt14.py#L79).

Basically our development set is not exactly the paper's dev set. We can choose to build one ourself cz we have the data building scripts but maybe later (hence opening this issue to document).

**Comment by ariG23498** [comment]
Thanks for bringing this up. Piggybacking on this line of thought, we should also probably think our tokenizer and embedding schemes a little more.

According to the paper, the embedding matrix is shared between the source and target languages. Also, the tokenizer shares the vocab -- which is very weird to me, but for the sake of reproducibility, we should probably update our code?

WDYT
